src/as/materials/MeshPhongMaterial.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ * @author alteredq / http://alteredqualia.com/
+ * @author Joe Pea / https://github.com/trusktr
+ */
+
+import {Color} from '../math/Color'
+import {Texture} from '../textures/Texture'
+import {Vector2} from '../math/Vector2'
+import {/*MaterialParameters,*/ Material} from './Material'
+import {Combine, NormalMapTypes, MultiplyOperation, TangentSpaceNormalMap} from '../constants'
+
+// This is soaking WET, let's not use this.
+// export interface MeshPhongMaterialParameters extends MaterialParameters {
+// 	/** geometry color in hexadecimal. Default is 0xffffff. */
+// 	color?: Color | string | f32
+// 	specular?: Color | string | f32
+// 	shininess?: f32
+// 	opacity?: f32
+// 	map?: Texture
+// 	lightMap?: Texture
+// 	lightMapIntensity?: f32
+// 	aoMap?: Texture
+// 	aoMapIntensity?: f32
+// 	emissive?: Color | string | f32
+// 	emissiveIntensity?: f32
+// 	emissiveMap?: Texture
+// 	bumpMap?: Texture
+// 	bumpScale?: f32
+// 	normalMap?: Texture
+// 	normalMapType?: NormalMapTypes
+// 	normalScale?: Vector2
+// 	displacementMap?: Texture
+// 	displacementScale?: f32
+// 	displacementBias?: f32
+// 	specularMap?: Texture
+// 	alphaMap?: Texture
+// 	envMap?: Texture
+// 	combine?: Combine
+// 	reflectivity?: f32
+// 	refractionRatio?: f32
+// 	wireframe?: boolean
+// 	wireframeLinewidth?: f32
+// 	wireframeLinecap?: string
+// 	wireframeLinejoin?: string
+// 	skinning?: boolean
+// 	morphTargets?: boolean
+// 	morphNormals?: boolean
+// }
+
+export class MeshPhongMaterial extends Material {
+	isMeshPhongMaterial: true = true
+
+	color: Color = new Color(0xffffff) // diffuse
+	specular: Color = new Color(0x111111)
+	shininess: f32 = 30
+
+	map: Texture | null = null
+
+	lightMap: Texture | null = null
+	lightMapIntensity: f32 = 1.0
+
+	aoMap: Texture | null = null
+	aoMapIntensity: f32 = 1.0
+
+	emissive: Color = new Color(0x000000)
+	emissiveIntensity: f32 = 1.0
+	emissiveMap: Texture | null = null
+
+	bumpMap: Texture | null = null
+	bumpScale: f32 = 1
+
+	normalMap: Texture | null = null
+	normalMapType: NormalMapTypes = TangentSpaceNormalMap
+	normalScale: Vector2 = new Vector2(1, 1)
+
+	displacementMap: Texture | null = null
+	displacementScale: f32 = 1
+	displacementBias: f32 = 0
+
+	specularMap: Texture | null = null
+
+	alphaMap: Texture | null = null
+
+	envMap: Texture | null = null
+	combine: Combine = MultiplyOperation
+	reflectivity: f32 = 1
+	refractionRatio: f32 = 0.98
+
+	wireframe: boolean = false
+	wireframeLinewidth: f32 = 1
+	wireframeLinecap: string = 'round'
+	wireframeLinejoin: string = 'round'
+
+	skinning: boolean = false
+	morphTargets: boolean = false
+	morphNormals: boolean = false
+
+	constructor(/*parameters?: MeshPhongMaterialParameters*/) {
+		super()
+
+		this.type = 'MeshPhongMaterial'
```

Comment:
i feel like we might need an an enum for these strings? thoughts?

Reply:
Too much to manage I think. Plus not sure if we even need to keep it, because these are used for type checks in plain Three.js, but in AS everything is so strictly typed we can rely on `instanceof` checks, and there is no duck typing.

In Three.js, you can make a plain object that is not `instanceof`, and just stick `type: 'MeshPhongMaterial'` with the properties that the consuming code expects, and it'll just work, but that's not possible in AS.

But for now, I think it's easier to leave it as is, until we decide if we'll remove them. Keeping them might make existing Three.js user code easier to port for people that switch to glas eventually.